basic/src/test/java/org/jboss/hal/testsuite/test/configuration/transactions/JournalTestCase.java

```diff
@@ -1,62 +1,67 @@
-    }
-
-    @Before
-    public void before() {
-        page.navigate();
-    }
-
     @Test
     public void toggleUseJournalStore() throws Exception {
-        ModelNode value = operations.readAttribute(TRANSACTIONS_ADDRESS, USE_JOURNAL_STORE_ATTR);
+        navigate2store();
+        boolean originalUseJournalStoreValue = isUseJournalStore();
         try {
             page.getConfigFragment().editCheckboxAndSave(USE_JOURNAL_STORE_ATTR, true);
             new ResourceVerifier(TRANSACTIONS_ADDRESS, client).verifyAttribute(USE_JOURNAL_STORE_ATTR, true);
             Console.withBrowser(browser).dismissReloadRequiredWindowIfPresent();
             page.getConfigFragment().editCheckboxAndSave(USE_JOURNAL_STORE_ATTR, false);
             new ResourceVerifier(TRANSACTIONS_ADDRESS, client).verifyAttribute(USE_JOURNAL_STORE_ATTR, false);
         } finally {
-            operations.writeAttribute(TRANSACTIONS_ADDRESS, USE_JOURNAL_STORE_ATTR, value);
+            writeUseJournalStore(originalUseJournalStoreValue);
         }
     }
 
     @Test
     public void toggleJournalStoreEnableAsyncIO() throws Exception {
-        page.getConfigFragment().editCheckboxAndSave(JOURNAL_STORE_ENABLE_ASYNC_IO_ATTR, true);
-        new ResourceVerifier(TRANSACTIONS_ADDRESS, client).verifyAttribute(JOURNAL_STORE_ENABLE_ASYNC_IO_ATTR, true);
-        Console.withBrowser(browser).dismissReloadRequiredWindowIfPresent();
-        page.getConfigFragment().editCheckboxAndSave(JOURNAL_STORE_ENABLE_ASYNC_IO_ATTR, false);
-        new ResourceVerifier(TRANSACTIONS_ADDRESS, client).verifyAttribute(JOURNAL_STORE_ENABLE_ASYNC_IO_ATTR, false);
+        boolean originalUseJournalStoreValue = isUseJournalStore();
+        try {
+            if (!originalUseJournalStoreValue) {
+                writeUseJournalStore(true);
+            }
+            navigate2store();
+            page.getConfigFragment().editCheckboxAndSave(JOURNAL_STORE_ENABLE_ASYNC_IO_ATTR, true);
+            new ResourceVerifier(TRANSACTIONS_ADDRESS, client).verifyAttribute(JOURNAL_STORE_ENABLE_ASYNC_IO_ATTR, true);
+            Console.withBrowser(browser).dismissReloadRequiredWindowIfPresent();
+            page.getConfigFragment().editCheckboxAndSave(JOURNAL_STORE_ENABLE_ASYNC_IO_ATTR, false);
+            new ResourceVerifier(TRANSACTIONS_ADDRESS, client).verifyAttribute(JOURNAL_STORE_ENABLE_ASYNC_IO_ATTR, false);
+        } finally {
+            writeUseJournalStore(originalUseJournalStoreValue);
+        }
+    }
+
+    private void navigate2store() {
+        page.navigate();
+        page.getConfig().switchTo("Store");
+    }
+
+    private boolean isUseJournalStore() throws IOException {
+        return operations.readAttribute(TRANSACTIONS_ADDRESS, USE_JOURNAL_STORE_ATTR).value().asBoolean();
+    }
+
+    private void writeUseJournalStore(boolean value) throws IOException, InterruptedException, TimeoutException {
+        if (isUseJournalStore() != value) {
```

Comment:
Handling for performance reasons? :-) Is it really needed, does the transaction subsystem require reload even when the value wasn't changed?